frontend/src/services/query/QueryServiceImpl.ts

```diff
@@ -56,55 +99,87 @@ export default class QueryServiceImpl extends QueryService {
     return this.http.post<QueryResult>(url, query, cancellation);
   }
 
-  // TODO: Cache entity details: https://github.com/amosproj/amos-ss2021-project2-context-map/issues/62
-
-  // TODO: This is currently not in use, so untested, add tests when this is needed in production code.
+  // TODO #249: Re-add ignored functions in QueryServiceImpl.ts to coverage when used #249
   /* istanbul ignore next */
   public async getEdgesById(
     idsOrDescriptors: number[] | EdgeDescriptor[],
     cancellation?: CancellationToken
   ): Promise<Edge[]> {
-    const batches = createBatches(idsOrDescriptors);
-
-    return (
-      await Promise.all(
-        batches.map((batch) =>
-          this.http.get<Edge[]>(
-            '/api/getEdgesById',
-            buildDetailsRequest(batch),
-            cancellation
-          )
-        )
-      )
-    ).flat();
+    return this.getEntitiesById(this.edgesById, idsOrDescriptors, cancellation);
   }
 
-  // TODO: This is currently not in use, so untested, add tests when this is needed in production code.
+  // TODO #249: Re-add ignored functions in QueryServiceImpl.ts to coverage when used #249
   /* istanbul ignore next */
   public async getNodesById(
     idsOrDescriptors: number[] | NodeDescriptor[],
     cancellation?: CancellationToken
   ): Promise<Node[]> {
-    const batches = createBatches(idsOrDescriptors);
-
-    return (
-      await Promise.all(
-        batches.map((batch) =>
-          this.http.get<Node[]>(
-            '/api/getNodesById',
-            buildDetailsRequest(batch),
-            cancellation
-          )
+    return this.getEntitiesById(this.nodesById, idsOrDescriptors, cancellation);
+  }
+
+  // TODO #249: Re-add ignored functions in QueryServiceImpl.ts to coverage when used #249
+  /* istanbul ignore next */
+  private getEntitiesById<T = Node | Edge>(
+    entitiesById: BatchCache<T>,
+    idsOrDescriptors: number[] | NodeDescriptor[],
+    cancellation?: CancellationToken
+  ): Promise<T[]> {
+    const cache = entitiesById.entities.getState();
+
+    // Ids as number.
+    const ids = idsOrDescriptors.map(
+      (idOrDescriptor: number | { id: number }) =>
+        typeof idOrDescriptor === 'number' ? idOrDescriptor : idOrDescriptor.id
+    );
+    // Already cached values.
+    // Stored here since they might get removed from the cache during the async operation.
+    const foundValues = ids
+      .filter((id) => cache.has(id))
```

Comment:
Revert map and filter operations, so you don't need to index twice.
`ids.map((id) => cache.get(id)).filter((e) => e !== undefined)`